tests/testthat/test-shinytest.R

```diff
@@ -0,0 +1,275 @@
+    testthat::expect_match(file_format_text, "pdf\n", fixed = TRUE)
+
+    testthat::expect_equal(
+      app$active_module_element_text("downbutton-file_name-label"),
+      "File name (without extension)"
+    )
+
+    testthat::expect_equal(
+      app$active_module_element("downbutton-file_name", hash = FALSE) %>%
+        app$get_value(input = .),
+      "table_20240322_161122"
+    )
+
+    pagination <- "downbutton-dwnl .paginate-ui .form-group.shiny-input-container"
+    pagination_class <- gsub("#", "#dropdown-menu-", app$active_module_element(pagination))
+    pagination_text <- app$get_text(pagination_class)
+    testthat::expect_match(pagination_text, "Paginate table:\n", fixed = TRUE)
+    testthat::expect_match(pagination_text, "lines / page\n", fixed = TRUE)
+
+    download_button <- app$get_html_rvest(app$active_module_element("downbutton-data_download > i"))
+    testthat::expect_equal(
+      download_button %>%
+        html_node("i") %>%
+        html_attr("class"),
+      "fas fa-download"
+    )
+    testthat::expect_equal(
+      download_button %>%
+        html_node("i") %>%
+        html_attr("aria-label"),
+      "download icon"
+    )
+
+    app$click(selector = "input[value='.csv']")
+    # check that pagination is missing
+    # app$get_text(pagination_class) # this returns values even though pagination is missing from the view
+    testthat::expect_false(
+      any(
+        unlist(
+          app$get_js(
+            sprintf(
+              "Array.from(document.querySelectorAll('%s')).map(el => el.checkVisibility())",
+              pagination_class
+            )
+          )
+        )
+      )
+    )
+
+    # Click the second TABLE button.
+    app$click(selector = app$active_module_element("expand"))
+    app$wait_for_idle(timeout = default_idle_timeout)
+    # Review the table modal content.
+
+    table_content <- app$active_module_element_text("table_out_modal")
+
+    check_table <- function(content) {
+      testthat::expect_match(content, "A: Drug X", fixed = TRUE)
+      testthat::expect_match(content, "C: Combination", fixed = TRUE)
+      testthat::expect_match(content, "UNDIFFERENTIATED", fixed = TRUE)
+      testthat::expect_match(content, "SEX", fixed = TRUE)
+      testthat::expect_match(content, "34.91", fixed = TRUE)
+      testthat::expect_match(content, "33.02", fixed = TRUE)
+    }
+    check_table(table_content)
+
+    # Close modal.
+    app$click(selector = "#shiny-modal-wrapper .modal-footer > button")
+    app$wait_for_idle(timeout = default_idle_timeout)
+    testthat::expect_null(app$get_html(app$active_module_element("table_out_modal")))
+
+    # Review the main table content.
+    main_table_content <- app$active_module_element_text("table_out_main")
+    check_table(main_table_content)
+
+    app$stop()
+  }
+)
+
+testthat::test_that(
+  "e2e: teal.widgets::plot_with_settings is initialized with proper plot and buttons",
```

Comment:
Please make sure that tests don't overlap with existing ones 
https://github.com/insightsengineering/teal.widgets/blob/3176ff54b7aef0e69528fad6dc4b6c51700c1539/tests/testthat/test-plot_with_settings_ui.R#L22

However, existing ones are nice to revisit so I think you'll just remove the previous ones.

Reply:
yes, there were a couple of simple tests already. I aim to add a bit more that verifies the structure of the output that is assumed to exist on a widget creation. The ones that existed assumed screenshot is equivalent to the assumed one, once you clicked a couple of buttons.